### ISSUE TITLE
Adding option to use C# DefineConstants when exporting

### ISSUE THREAD

**Issue by SlejmUr** [godotengine/godot-proposals#9694]
### Describe the project you are working on

A game that is focusing on moddability, but has a separate server/client code

### Describe the problem or limitation you are having in your project

When exporting my application I tend to use C# constants that I define inside the .csproj and in code (.cs) looks like this:
```csharp
#if ServerCode
..
#elif ClientCode
...
#endif
```

And inside the .csproj file is:
```xml
<DefineConstants>$(DefineConstants);DRMFREE;CLIENT</DefineConstants>
<!--DefineConstants>$(DefineConstants);DRMFREE;SERVER</DefineConstants-->
<!--DefineConstants>$(DefineConstants);DRMFREE;CLIENT;DEMO</DefineConstants-->
.. (more define constants for individual exports, ie: steam, demo, etc)
```

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I dont need to individually edit these when exporting into different content. And could be automated with only just exporting

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I already have a working code that it is [here](https://github.com/ExtractIntoVoid/godot)

But basicly it is creating a same option when exporting (like the dotnet/embed_build_outputs)
Getting the option, checking if not null or empty, then proceed to sanitize the string.
It would then add into custom properties GodotExportConstants.
In argument it would look like:

/p:GodotExportConstants=DRMFREE%3BCLIENT

It would also add into GodotDefineConstants then would be just added to DefineConstants

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I havent seen anything that would fire before the building/publishing process starts and since Features arent exposed to dotnet it would be impossible to gather what features are made.

### Is there a reason why this should be core and not an add-on in the asset library?

Asset library doesnt contains anything for exporting something like this. (Or I havent seen like that)

**Comment by xill47** [comment]
I've been working around this by setting environmental variable before doing export, but it would be 100% more convenient to have this built-in into export_presets.cfg

**Comment by raulsntos** [comment]
- Sounds like https://github.com/godotengine/godot/pull/53920 would fulfill your use case.

**Comment by SlejmUr** [comment]
Exactly, but without the starting of GODOT_FEATURE_ and only focusing to define constants.

**Comment by paulloz** [comment]
> * Sounds like [Re-add project features as define constants in C# godot#53920](https://github.com/godotengine/godot/pull/53920) would fulfill your use case.

I don't think so, the user scenario here would be (as far as I understand) "I want to be able to specify custom constants to be used when Godot exports my game".
